gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -319,22 +321,22 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
           //   and need some tests before being enabled.
           TransformHints.tagNotTransformable(plan)
         case plan: ProjectExec =>
-          if (!enableColumnarProject) {
+          if (!enableColumnarProject || scanOnly) {
```

Comment:
Can we move scanOnly when we set "enableColumnarProject"? Which means the scan only option will overwrite all other operator's enablexxx config.

Reply:
Does that mean putting `scanOnly` before `!enableColumnarProject`?

Reply:
No, use something like 
enableColumnarProject = !scanonly && config.get(xxx)

Reply:
Fixed.